### PR TITLE
[JENKINS-69850]: Queue.maintain() does no longer trigger an infinite recursive loop after loading a queue from xml

### PR DESCRIPTION
**Pull request by Si-So**
See [JENKINS-69850](https://issues.jenkins.io/browse/JENKINS-69850).

### Testing done

QueueTest.handleCauseOfBlockageThatIsNull()

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@res0nance
@jtnord 
@basil 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7273"></a>

### CODE REVIEW

**Comment by Si-So** [comment]
This commit only includes a test for the issue. I expect only my new test to fail.

**Comment by Si-So** [comment]
On the Linux machines only my new test failed (as predicted). 
The Windows machine executed no tests at all. It looks like a problem on ci.jenkins.io. 

I will now push my commit and expect no test failures. 

FYI: I will be away from my computer until Monday and thus I will not be able to react to change requests.

File: core/src/main/java/hudson/model/Queue.java
Comment (basil) [inline core/src/main/java/hudson/model/Queue.java:2629]: The fix itself looks good. Thanks for the quick response! On a procedural note, a new public API either needs to be documented or restricted, as described in the **Submitter checklist.** In this case the only consumer is in core and I do not think we anticipate additional external consumers, so I would lean toward restricting the new API with `@Restricted(NoExternalUse.class)`.

Comment (Si-So) [reply core/src/main/java/hudson/model/Queue.java:2630]: Done

**Review by basil — APPROVED** [review]
Thanks again for the quick fix!

**Review by res0nance — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by Si-So** [comment]
> Thanks again for the quick fix!

Thank you for thanking me :)

**Pull request merged**
